exonum-java-binding/core/rust/integration_tests/tests/runtime.rs

```diff
@@ -0,0 +1,667 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+extern crate exonum_testkit;
+extern crate futures;
+extern crate hex;
+extern crate integration_tests;
+extern crate java_bindings;
+#[macro_use]
+extern crate lazy_static;
+extern crate tempfile;
+extern crate tokio_core;
+
+use exonum_testkit::TestKitBuilder;
+use futures::sync::mpsc;
+use hex::FromHex;
+use integration_tests::{
+    fake_runtime::create_fake_service_runtime_adapter,
+    fake_service::{
+        self, create_before_commit_test_map, create_init_service_test_map,
+        create_service_artifact_non_instantiable_service, create_service_artifact_non_loadable,
+        create_service_artifact_valid, create_tx_test_entry,
+    },
+    vm::{create_vm_for_tests_with_fake_classes, log4j_path},
+};
+use java_bindings::{
+    create_service_runtime,
+    exonum::{
+        blockchain::{Blockchain, ExecutionError, ExecutionErrorKind, InstanceConfig},
+        crypto::{gen_keypair, hash, Hash, PublicKey, SecretKey},
+        messages::{AnyTx, Verified},
+        node::ApiSender,
+        runtime::{ArtifactId, CallInfo, InstanceId, InstanceSpec, MethodId, Runtime},
+    },
+    exonum_merkledb::{BinaryValue, TemporaryDB},
+    jni::JavaVM,
+    utils::any_to_string,
+    DeployArguments, Error, Executor, JavaRuntimeProxy, RuntimeConfig,
+};
+use std::{panic, path::PathBuf, sync::Arc};
+use tempfile::TempPath;
+use tokio_core::reactor::Core;
+
+lazy_static! {
+    static ref VM: Arc<JavaVM> = create_vm_for_tests_with_fake_classes();
+}
+
+const VALID_INSTANCE_ID: u32 = 127;
+const VALID_INSTANCE_NAME: &str = "artifact__test_valid";
+const VALID_ARTIFACT_NAME: &str = "artifact:test-valid";
+const VALID_ARTIFACT_VERSION: &str = "1.0";
+
+#[test]
+fn runtime_exception_handling_checked_exception() {
+    let artifact = ArtifactId::new(JavaRuntimeProxy::RUNTIME_ID, "artifact:test-name:1.0").unwrap();
+    let mut runtime = get_fake_runtime("createRuntimeThrowingExceptions");
+    let future = runtime.deploy_artifact(artifact, vec![]);
+
+    let mut core = Core::new().unwrap();
+    let result = core.run(future);
+
+    assert!(result.is_err());
+
+    let error = result.unwrap_err();
+    let message = error.description;
+    let kind = error.kind;
+
+    match kind {
+        ExecutionErrorKind::Runtime { code } => {
+            assert_eq!(code, Error::JavaException as u8);
+            assert!(message.contains("deployArtifact"));
+            assert!(message.contains("ServiceLoadingException"))
+        }
+        _ => panic!("error.kind is not 'ExecutionErrorKind::Runtime'"),
+    }
+
+    runtime.shutdown();
+}
+
+#[test]
+fn runtime_exception_handling_unchecked_exception() {
```

Comment:
Unclear what is being tested: a particular operation (is_artifact_deployed)
or handling exceptions in a particular way?

Even if there are some other similar operations that is an implementation detail if they share the same code to turn some exceptions into panics. Hence I suggest:
1. Naming this test accordingly: is_artifact_deployed_shall_panic_on_unchecked_exceptions
2. If needed, testing any utilities used to handle exceptions separately